### ISSUE TITLE
"Unrecognized lexicon type" for every record in my BlueSky download

### ISSUE THREAD

**Issue by bobg** [bluesky-social/indigo#847]
Hello! I am trying to parse the contents of the `repo.car` file that I downloaded from BlueSky just now, using the sample code at https://docs.bsky.app/blog/repo-export#parse-records-from-car-file-as-json. I'm getting a [util.ErrUnrecognizedType](https://pkg.go.dev/github.com/bluesky-social/indigo/lex/util#pkg-variables) from [Repo.GetRecord](https://pkg.go.dev/github.com/bluesky-social/indigo/repo#Repo.GetRecord) for every record in the archive.

Here's my code. Am I missing a step? Thanks in advance.

```go
	...
	r, err := repo.ReadRepoFromCar(ctx, os.Stdin)
	if err != nil {
		return errors.Wrap(err, "reading repo from stdin")
	}

	err = r.ForEach(ctx, "", func(k string, v cid.Cid) error {
		_, rec, err := r.GetRecord(ctx, k)
		if errors.Is(err, util.ErrUnrecognizedType) {
			log.Printf("Skipping record %s: %s", k, err)
			return nil
		}
		...
	}
	...
```

**Comment by bnewbold** [comment]
Yup! Many of the default methods in the indigo codebase assume that there are defined structs/types for the records being marshalled/unmarshalled. This "registration" happens when relevant packages (like the `api/bsky` package) are imported.

It is also possible to pull out data as bytes, and unmarshall to generic `map[string]any` objects, using the `atproto/data` package. Check out the goat code for examples, eg:
https://github.com/bluesky-social/indigo/blob/main/cmd/goat/repo.go#L239

definitely need docs around all this, and probably refactor things to default to "schema/type agnostic" with the registered types being more explicit?

**Comment by bobg** [comment]
Thank you!
